src/dstruct/factor_graph/factor_graph.cpp

```diff
@@ -194,16 +181,37 @@ void dd::FactorGraph::load(const CmdParser & cmd, const bool is_quiet, int inc){
   // NOTE This is very important, as read_edges assume variables,
   // factors and weights are ordered so that their id is the index 
   // where they are stored in the array
-  this->sort_by_id();
+  if (!inc) this->sort_by_id(inc);
 
-  // load edges
-  n_loaded = read_edges(filename_edges, *this);
+  // load factors
+  n_loaded = read_factors(filename_factors, *this, inc);
   if(cmd.delta_folder->getValue() != ""){
-    std::cout << "Loading delta..." << std::endl;
-    n_loaded += read_edges(cmd.delta_folder->getValue() + "/graph.edges", *this);
+    std::cout << "Loading delta..." << cmd.delta_folder->getValue() + "/graph.factors" << std::endl;
+    n_loaded += read_factors(cmd.delta_folder->getValue() + "/graph.factors", *this, inc);
   }
+
+  assert(n_loaded == n_factor);
   if (!is_quiet) {
-    std::cout << "LOADED EDGES: #" << n_loaded << std::endl;
+    std::cout << "LOADED FACTORS: #" << n_loaded << std::endl;
+  }
+
+  // sort the above components
+  // NOTE This is very important, as read_edges assume variables,
+  // factors and weights are ordered so that their id is the index 
+  // where they are stored in the array
+  if (inc) this->sort_by_id(inc);
```

Comment:
I think it's better to refactor to make the workflow clearer. The `sort_by_id (inc)` is a bit ad hoc. The logic after reading variables and weights is straight-forward:
non-inc: 1. sort variables and weights by id 2. read factors
inc: 1. read factors 2. sort variables, weights, and factors by id 3. read edges

How about the following?
```
read_variables()
read_weights()
sort_variables()
sort_weights()

if (non-inc) {
  read_factors()
} else {
  read_factors_inc()
  sort_factors()
  read_edges_inc()}
```

Also, it is helpful to have comments on what is the difference in loading inc and non-inc.